rclcpp/include/rclcpp/any_subscription_callback.hpp

```diff
@@ -82,6 +88,38 @@ struct AnySubscriptionCallback
   {
     shared_ptr_with_info_callback = callback;
   }
+
+  template<typename CallbackT,
+  typename std::enable_if<
+    function_traits<CallbackT>::arity == 1
+  >::type * = nullptr,
+  typename std::enable_if<
+    std::is_same<
+      typename function_traits<CallbackT>::template argument_type<0>,
+      typename std::shared_ptr<const MessageT>
+    >::value
+  >::type * = nullptr
+  >
+  void set(CallbackT callback)
+  {
+    const_shared_ptr_callback = callback;
+  }
+
+  template<typename CallbackT,
+  typename std::enable_if<
+    function_traits<CallbackT>::arity == 2
+  >::type * = nullptr,
+  typename std::enable_if<
+    std::is_same<
+      typename function_traits<CallbackT>::template argument_type<0>,
+      typename std::shared_ptr<const MessageT>
+    >::value
+  >::type * = nullptr
+  >
+  void set(CallbackT callback)
+  {
```

Comment:
It'll work on Windows as is. The only thing is that the type of the second argument of the callback is not being checked. Unfortunately, we can't use `std::is_same...` for the second argument because of the lack of expression SFINAE in VS2015, so instead we can use `static_assert` in the body of the function to check the type.